### ISSUE TITLE
puzzle over optimizer_cfg setting

### ISSUE THREAD

**Issue by JoyLuo** [open-mmlab/mmsegmentation#698]
optimizer only use the cfg.optimizer, not use the optimizer_cfg
so, the optimizer_cfg is useless

v0.15.1 mmseg/apis/train.py 72 line
`optimizer = build_optimizer(model, cfg.optimizer)`

**Comment by sennnnn** [comment]
Yeah, the optimizer config is used to build optimizer in mmcv runner hook. You can check runner.register_training_hooks(cfg.lr_config, cfg.optimizer_config, cfg.checkpoint_config, cfg.log_config, cfg.get('momentum_config', None). We don't support use optimizer_config to build optimizer by mmcv hooks now.

**Comment by Junjun2016** [comment]
`optimizer` is used for building optimizer, while `optimizer_cfg` is used for building optimizer hook.

The optimizer hook is used for wrapping the optimizer which determines where to call optimizer and add more functions, e.g. `grad_clip`.
